### ISSUE TITLE
Unable to show events from December

### ISSUE THREAD

**Issue by mrfanatic** [jhonis/e-calendar#1]
If we have events on December, it can't show. If we pass events to the next year of January it shows again.

**Comment by EdithF86** [comment]
(function ($) {

```
var eCalendar = function (options, object) {
    // Initializing global variables
    var adDay = new Date().getDate();
    var adMonth = new Date().getMonth();
    var adYear = new Date().getFullYear();
    var dDay = adDay;
    var dMonth = adMonth;
    var dYear = adYear;
    var instance = object;

    var settings = $.extend({}, $.fn.eCalendar.defaults, options);

    function lpad(value, length, pad) {
        if (typeof pad == 'undefined') {
            pad = '0';
        }
        var p;
        for (var i = 0; i < length; i++) {
            p += pad;
        }
        return (p + value).slice(-length);
    }

    var mouseOver = function () {
        $(this).addClass('c-nav-btn-over');
    };
    var mouseLeave = function () {
        $(this).removeClass('c-nav-btn-over');
    };
    var mouseOverEvent = function () {
        $(this).addClass('c-event-over');
        var d = $(this).attr('data-event-day');
        $('div.c-event-item[data-event-day="' + d + '"]').addClass('c-event-over');
    };
    var mouseLeaveEvent = function () {
        $(this).removeClass('c-event-over')
        var d = $(this).attr('data-event-day');
        $('div.c-event-item[data-event-day="' + d + '"]').removeClass('c-event-over');
    };
    var mouseOverItem = function () {
        $(this).addClass('c-event-over');
        var d = $(this).attr('data-event-day');
        $('div.c-event[data-event-day="' + d + '"]').addClass('c-event-over');
    };
    var mouseLeaveItem = function () {
        $(this).removeClass('c-event-over')
        var d = $(this).attr('data-event-day');
        $('div.c-event[data-event-day="' + d + '"]').removeClass('c-event-over');
    };
    var nextMonth = function () {
        if (dMonth < 11) {
            dMonth++;
        } else {
            dMonth = 0;
            dYear++;
        }
        print();
    };
    var previousMonth = function () {
        if (dMonth > 0) {
            dMonth--;
        } else {
            dMonth = 11;
            dYear--;
        }
        print();
    };

    function loadEvents() {
        if (typeof settings.url != 'undefined' && settings.url != '') {
            $.ajax({url: settings.url,
                async: false,
                success: function (result) {
                    settings.events = result;
                }
            });
        }
    }

    function print() {
        loadEvents();
        var dWeekDayOfMonthStart = new Date(dYear, dMonth, 1).getDay();
        var dLastDayOfMonth = new Date(dYear, dMonth + 1, 0).getDate();
        var dLastDayOfPreviousMonth = new Date(dYear, dMonth + 1, 0).getDate() - dWeekDayOfMonthStart + 1;

        var cBody = $('<div/>').addClass('c-grid');
        var cEvents = $('<div/>').addClass('c-event-grid');
        var cEventsBody = $('<div/>').addClass('c-event-body');
        cEvents.append($('<div/>').addClass('c-event-title c-pad-top').html(settings.eventTitle));
        cEvents.append(cEventsBody);
        var cNext = $('<div/>').addClass('c-next c-grid-title c-pad-top');
        var cMonth = $('<div/>').addClass('c-month c-grid-title c-pad-top');
        var cPrevious = $('<div/>').addClass('c-previous c-grid-title c-pad-top');
        cPrevious.html(settings.textArrows.previous);
        cMonth.html(settings.months[dMonth] + ' ' + dYear);
        cNext.html(settings.textArrows.next);

        cPrevious.on('mouseover', mouseOver).on('mouseleave', mouseLeave).on('click', previousMonth);
        cNext.on('mouseover', mouseOver).on('mouseleave', mouseLeave).on('click', nextMonth);

        cBody.append(cPrevious);
        cBody.append(cMonth);
        cBody.append(cNext);
        for (var i = 0; i < settings.weekDays.length; i++) {
            var cWeekDay = $('<div/>').addClass('c-week-day c-pad-top');
            cWeekDay.html(settings.weekDays[i]);
            cBody.append(cWeekDay);
        }
        var day = 1;
        var dayOfNextMonth = 1;
        for (var i = 0; i < 42; i++) {
            var cDay = $('<div/>');
            if (i < dWeekDayOfMonthStart) {
                cDay.addClass('c-day-previous-month c-pad-top');
                cDay.html(dLastDayOfPreviousMonth++);
            } else if (day <= dLastDayOfMonth) {
                cDay.addClass('c-day c-pad-top');
                if (day == dDay && adMonth == dMonth && adYear == dYear) {
                    cDay.addClass('c-today');
                }
                for (var j = 0; j < settings.events.length; j++) {
                    var d = settings.events[j].datetime;
                    var mon = d.getMonth() - 1;
                    var yr = d.getFullYear();
                    if (d.getMonth() == 0) {
                        mon = 11;
                        yr = d.getFullYear() - 1;
                    }

                    if (d.getDate() == day && mon == dMonth && yr == dYear) {
                        cDay.addClass('c-event').attr('data-event-day', d.getDate());
                        cDay.on('mouseover', mouseOverEvent).on('mouseleave', mouseLeaveEvent);
                    }
                }
                cDay.html(day++);
            } else {
                cDay.addClass('c-day-next-month c-pad-top');
                cDay.html(dayOfNextMonth++);
            }
            cBody.append(cDay);
        }
        var eventList = $('<div/>').addClass('c-event-list');
        for (var i = 0; i < settings.events.length; i++) {
            var d = settings.events[i].datetime;
            var mon = d.getMonth() - 1;
            var yr = d.getFullYear();
            if (d.getMonth() == 0) {
                mon = 11;
                yr = d.getFullYear() - 1;
            }

            if (mon == dMonth && yr == dYear) {
                var date = lpad(d.getDate(), 2) + '/' + lpad(d.getMonth(), 2) + ' ' + lpad(d.getHours(), 2) + ':' + lpad(d.getMinutes(), 2);
                var item = $('<div/>').addClass('c-event-item');
                var title = $('<div/>').addClass('title').html(date + '  ' + settings.events[i].title + '<br/>');
                var description = $('<div/>').addClass('description').html(settings.events[i].description + '<br/>');
                item.attr('data-event-day', d.getDate());
                item.on('mouseover', mouseOverItem).on('mouseleave', mouseLeaveItem);
                item.append(title).append(description);
                eventList.append(item);
            }
        }
        $(instance).addClass('calendar');
        cEventsBody.append(eventList);
        $(instance).html(cBody).append(cEvents);
    }

    return print();
}

$.fn.eCalendar = function (oInit) {
    return this.each(function () {
        return eCalendar(oInit, $(this));
    });
};

// plugin defaults
$.fn.eCalendar.defaults = {
    weekDays: ['Dom', 'Seg', 'Ter', 'Qua', 'Qui', 'Sex', 'Sab'],
    months: ['Janeiro', 'Fevereiro', 'Março', 'Abril', 'Maio', 'Junho', 'Julho', 'Agosto', 'Setembro', 'Outubro', 'Novembro', 'Dezembro'],
    textArrows: {previous: '<', next: '>'},
    eventTitle: 'Eventos',
    url: '',
    events: [
        {title: 'Brasil x Croácia', description: 'Abertura da copa do mundo 2014', datetime: new Date(2014, 6, 12, 17)},
        {title: 'Brasil x México', description: 'Segundo jogo da seleção brasileira', datetime: new Date(2014, 6, 17, 16)},
        {title: 'Brasil x Camarões', description: 'Terceiro jogo da seleção brasileira', datetime: new Date(2014, 6, 23, 16)}
    ]
};
```

}(jQuery));

**Comment by FPetruzzo** [comment]
This new code fixed the issue with December, but I noticed after that when displaying the current month, the days of the week from the previous month were not correct.  I believe I was able to fix that by adjusting this variable in the code:
 var dLastDayOfPreviousMonth = new Date(dYear, dMonth + 1, 0).getDate() - dWeekDayOfMonthStart + 1;

to 
 var dLastDayOfPreviousMonth = new Date(dYear, dMonth, 0).getDate() - dWeekDayOfMonthStart + 1;

Works great by the way, thanks for the widget!

**Comment by jhonis** [comment]
Aparently this error was fixed in the build 0.9.2 (issue #4). If the error keep occuring please open another issue.
